### ISSUE TITLE
tldr - Oudated spec.

### ISSUE THREAD

**Issue by seenukarthi** [withfig/autocomplete#1826]
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [X] I would like to work on this.

### What CLI tool does this relate to?

tldr

### Which statement makes the most sense?

There is a bug with this completion spec

### Issue Details

The `tldr` spec is outdated.

1. The local cache folder needs to be corrected. It is moved from `~/.tldr/cache` to `~/.tldrc/tldr`.
2. `-p, --platform` option is missing.
3. The following options are no longer supported.
   * `-a, --list-all`
   * `--random`
   * `--random-example`
   * `-m, --markdown`
